Sources/SKSupport/Collection.swift

```diff
@@ -10,7 +10,23 @@
 //
 //===----------------------------------------------------------------------===//
 
-extension RandomAccessCollection where Element: Equatable {
+extension RandomAccessCollection where Element: Equatable & Hashable {
+
+  /// Searches back through both patterns until we find an element that
```

Comment:
s/both patterns/self and the given pattern/ ?

Reply:
Could you give me more details?

Reply:
There is only one pattern.  It searches in `pattern` and `self`, not in two patterns.